apps/ejabberd/src/mongoose_api_common.erl

```diff
@@ -351,3 +357,35 @@ method_to_action(<<"POST">>) -> create;
 method_to_action(<<"PUT">>) -> update;
 method_to_action(<<"DELETE">>) -> delete.
 
+%%--------------------------------------------------------------------
+%% Authorization
+%%--------------------------------------------------------------------
+
+get_creds(Req) ->
+    case get_auth_details(Req) of
+        {ok, undefined, _} ->
+           undefined;
+        {ok, {_AuthMethod, Creds}, _Req2} ->
```

Comment:
`get_creds` and `get_auth_method` are never used separately. Why not merge them? Or remove them and have `get_auth_details` return `undefined | {AuthMethod, Creds}` and use it in respective API modules.